server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala

```diff
@@ -188,3 +193,20 @@ object AbfsFileSigner {
       getRelativePath(abfsStore, _))
   }
 }
+
+class GCSFileSigner(
+    name: URI,
+    conf: Configuration,
+    preSignedUrlTimeoutSeconds: Long) extends CloudFileSigner{
+
+  override def sign(path: Path): String = {
+    val absPath = path.toUri
+    val bucketName = absPath.getHost
+    val objectName = absPath.getPath.stripPrefix("/")
+    val storage = StorageOptions.newBuilder.build.getService
```

Comment:
Is `storage` thread-safe? If so, we can save it as a val in `GCSFileSigner` to avoid loading the credentials for each url.

Reply:
I think `storage` is thread-safe.
One instance of `GCSFileSigner` is created in each request for client, so each thread for a request has a dedicated instance of `GCSFileSigner` and won't touch other `GCSFileSigner` instances.
And, `storage` is used only for reading, so it will be OK if some threads for creating a presignied-URL are processed in paralle.